src/oscar/apps/dashboard/nav.py

```diff
@@ -86,7 +86,7 @@ def default_access_fn(user, url_name, url_args=None, url_kwargs=None):
     # label that can be loaded by get_class (e.g.
     # 'dashboard.catalogue.app), which then essentially checks
     # INSTALLED_APPS for the right module to load
-    match = re.search('(dashboard[\w\.]*)\.views$', view_module)
+    match = re.search('(dashboard[\w\.]*)\.\w*$', view_module)
```

Comment:
What is the exact consequence of this change? What is the reactio to the actio?

Reply:
I updated the description of this PR, I hope this answers your question?

I also changed `\w*` to `\w+`, which prevents an empty view's module name to be matched, which should not happen in reality anyhow.

Reply:
Ok, so in short:

Before: view in module 'views' in some module X, the path of which starts with 'dashboard', application in module 'app' in the same module X.

After: view in any module in some module X, the path of which starts with 'dashboard', application in module 'app' in the same module X.

Note: if module X has the 'app' module in it, views that are to be used in the navigation still cannot be in a module that's not directly in the module X. In other words, `X.some_module.a_view_function` is possible, but `X.some_module.some_submodule.a_view_function` is not.

Reply:
Exactly.